### ISSUE TITLE
StreamServerEndpointService doesn't log listen errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4731]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4731|
|Type|release blocker: regression|
|Created|2010-11-06 01:05:04Z|
|Branch|https://github.com/twisted/twisted/tree/cantlisten-4731|

This program (run unprivileged, so the port cannot be bound) exhibits incorrect behavior:

```
#!py
from twisted.internet.endpoints import serverFromString
from twisted.application.internet import StreamServerEndpointService
from twisted.internet import reactor
from twisted.internet.protocol import Factory
from twisted.application.service import Application

application = Application("foo")
svc = StreamServerEndpointService(
    serverFromString(reactor, "tcp:80"), Factory())
svc.setServiceParent(application)
```

<details><summary>Searchable metadata</summary>

```
trac-id__4731 4731
type__release_blocker__regression release blocker: regression
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_cantlisten_4731 branches/cantlisten-4731
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1289005504000000 1289005504000000
changetime__1289463981000000 1289463981000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

Here's the example expressed using APIs available on 10.1, to make the regression clearer:

```
#!py
from twisted.application.strports import service
from twisted.internet.protocol import Factory
from twisted.application.service import Application

application = Application("foo")
svc = service("tcp:80", Factory())
svc.setServiceParent(application)
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

(In [30245]) Branching to 'cantlisten-4731'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph removed owner|
|-|-|

[Build Results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/cantlisten-4731).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @glyph|
|-|-|

This looks good.  It needs a news fragment, then please merge.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set status to `closed`|
|-|-|

(In [30265]) 
Correct behavior of StreamServerEndpointService and strports.service().

Author: glyph

Reviewer: exarkun

Fixes: [#4731](https://github.com/twisted/twisted/issues/4731)

This corrects a regression where StreamServerEndpointService.startService() would never log an error if could not actually listen on its designated endpoint.  Also, strports.service().startService() now raises an exception (as it used to) which will cause twistd to exit in some cases, because there is still no other, better API to halt service initialization.  However, StreamServerEndpointService itself, if constructed directly, will not inherit this (accidental) behavior.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
